src/Nethermind/Nethermind.Consensus.AuRa/Transactions/TxPermissionFilter.cs

```diff
@@ -22,25 +22,28 @@
 using Nethermind.Core.Crypto;
 using Nethermind.Dirichlet.Numerics;
 using Nethermind.Logging;
+using Nethermind.State;
 
 namespace Nethermind.Consensus.AuRa.Transactions
 {
     public class TxPermissionFilter : ITxPermissionFilter
     {
         private readonly TransactionPermissionContract _contract;
         private readonly ITxPermissionFilter.Cache _cache;
+        private readonly IStateProvider _stateProvider;
         private readonly ILogger _logger;
         
-        public TxPermissionFilter(TransactionPermissionContract contract, ITxPermissionFilter.Cache cache, ILogManager logManager)
+        public TxPermissionFilter(TransactionPermissionContract contract, ITxPermissionFilter.Cache cache, IStateProvider stateProvider, ILogManager logManager)
         {
             _contract = contract ?? throw new ArgumentNullException(nameof(contract));
             _cache = cache ?? throw new ArgumentNullException(nameof(cache));
+            _stateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
             _logger = logManager?.GetClassLogger<TxPermissionFilter>() ?? throw new ArgumentNullException(nameof(logManager));
         }
         
-        public bool IsAllowed(Transaction tx, BlockHeader blockHeader)
+        public bool IsAllowed(Transaction tx, BlockHeader blockHeader, long blockNumber)
         {
-            if (blockHeader.Number + 1 < _contract.ActivationBlock)
+            if (blockNumber < _contract.ActivationBlock)
```

Comment:
it was better before

Reply:
Not really, there are 2 cases: We create a block and do it against parent, or we check new before processing against current (it would actually be hard to find parent). Thus this is more complicated but better.